src/shared/components/match/match-score.tsx

```diff
@@ -18,20 +18,60 @@ interface MatchScoreProps {
 export const MatchScore = React.memo<MatchScoreProps>(
   ({ match, children }: MatchScoreProps) => {
     const { formatDate, formatTime } = useDate();
+    const today = new Date();
+    const tomorrow = new Date();
+    tomorrow.setDate(today.getDate() + 1);
+    const yesterday = new Date();
+    yesterday.setDate(today.getDate() - 1);
 
     const styles = getStyles(styleTokens);
 
     const translate = useTranslate();
 
     const navigation = useNavigation();
 
+    const todayDay = today.getDate();
+    const todayYear = today.getFullYear();
+    const todayMonthAbbreviation = new Intl.DateTimeFormat('en-US', { month: 'short' }).format(
+      today
+    );
+
+    const tomorrowDay = tomorrow.getDate();
+    const tomorrowYear = tomorrow.getFullYear();
+    const tomorrowMonthAbbreviation = new Intl.DateTimeFormat('en-US', { month: 'short' }).format(
+      tomorrow
+    );
+
+    const yesterdayDay = yesterday.getDate();
+    const yesterdayYear = yesterday.getFullYear();
+    const yesterdayMonthAbbreviation = new Intl.DateTimeFormat('en-US', { month: 'short' }).format(
+      yesterday
+    );
+
+    const tomorrowString = `${tomorrowDay} ${tomorrowMonthAbbreviation} ${tomorrowYear}`;
+    const todayString = `${todayDay} ${todayMonthAbbreviation} ${todayYear}`;
+    const yesterdayString = `${yesterdayDay} ${yesterdayMonthAbbreviation} ${yesterdayYear}`;
+
+    const formattedDate = formatDate(match.date);
+    const formattedTime = formatTime(match.date);
+
+    let dateLabel = formattedDate;
+
+    if (formattedDate === todayString) {
+      dateLabel = translate('home.today');
+    } else if (formattedDate === tomorrowString) {
+      dateLabel = translate('home.tomorrow');
+    } else if (formattedDate === yesterdayString) {
+      dateLabel = translate('home.yesterday');
+    }
```

Comment:
Instead of comparing string, you can compare the date with `isSameDay`. Then, you will not need `todayDay`, `tomorrowMonthAbbreviation`, `todayYear`, `todayString`, etc. https://github.com/wanned/karmine-corp-app/blob/b4082dd734def341f6ed4da5edc5453025acd214/src/shared/utils/is-same-day.ts